src/commands/deploy.ts

```diff
@@ -287,78 +287,99 @@ async function runPreDeployTask(deployFsPath: string, telemetryProperties: Telem
         }
     }
 
     if (preDeployTask) {
         telemetryProperties.foundPreDeployTask = 'true';
         await vscode.tasks.executeTask(preDeployTask);
-        await new Promise((resolve: () => void, reject: (error: Error) => void): void => {
-            const listener: vscode.Disposable = vscode.tasks.onDidEndTaskProcess((e: vscode.TaskProcessEndEvent) => {
-                if (e.execution.task === preDeployTask) {
-                    listener.dispose();
-                    if (e.exitCode === 0) {
-                        resolve();
-                    } else {
-                        reject(new Error(localize('taskFailed', 'Failed to deploy. Pre-deploy task "{0}" failed with exit code "{1}".', e.execution.task.name, e.exitCode)));
-                    }
-                }
-            });
-        });
+        // tslint:disable-next-line:no-non-null-assertion
+        await waitForPreDeployTask(preDeployTask, telemetryProperties, workspaceFolderPath!);
     } else {
         telemetryProperties.foundPreDeployTask = 'false';
 
         const messageLines: string[] = [];
         // If the task name was specified in the user's settings, we will throw an error and block the user's deploy if we can't find that task
         // If the task name was _not_ specified, we will display a warning and let the deployment continue. (The preDeployTask isn't _always_ necessary and we don't want to block old projects that never had this setting)
-        if (isTaskNameDefinedInSettings) {
+        if (taskName) {
             messageLines.push(localize('noPreDeployTaskError', 'Did not find preDeploy task "{0}". Change the "{1}.{2}" setting, manually edit your task.json, or re-initialize your VS Code config with the following steps:', taskName, extensionPrefix, preDeployTaskSetting));
+            const fullMessage: string = getFullPreDeployMessage(messageLines);
+            throw new Error(fullMessage);
         } else {
-            messageLines.push(localize('noPreDeployTaskWarning', 'WARNING: Did not find preDeploy task "{0}". The deployment will continue, but the selected folder may not reflect your latest changes.', taskName));
-            messageLines.push(localize('howToAddPreDeploy', 'In order to ensure that you always deploy your latest changes, add a preDeploy task with the following steps:'));
+            const recommendedTaskName: string | undefined = getRecommendedTaskName(language, runtime);
+            if (recommendedTaskName) {
+                messageLines.push(localize('noPreDeployTaskWarning', 'WARNING: Did not find recommended preDeploy task "{0}". The deployment will continue, but the selected folder may not reflect your latest changes.', recommendedTaskName));
+                messageLines.push(localize('howToAddPreDeploy', 'In order to ensure that you always deploy your latest changes, add a preDeploy task with the following steps:'));
+                const fullMessage: string = getFullPreDeployMessage(messageLines);
```

Comment:
Why have a function to add these lines when you push the previous two lines into the array above?  I don't see `getFullPreDeployMessage` called anywhere else and it seems like these will always be called together.

Reply:
`getFullPreDeployMessage` is also called a few lines up. Just search for the function name

Reply:
This is good, however you must deploy the extensions for Blob Trigger and others or the runtime will fail.  Creating an azure function from scratch Azure Blob trigger and deploying per the document will result in failure   alias jsanders

Reply:
@jsandersrocks I don't quite understand what you're saying here. If you have an issue or would like a change in behavior please file a new issue and we can discuss further there: https://github.com/Microsoft/vscode-azurefunctions/issues/new